spec/models/waste_carriers_engine/registration_spec.rb

```diff
@@ -452,14 +452,22 @@ module WasteCarriersEngine
           end
 
           context "when the registration expiration date is today" do
-            let(:registration) { build(:registration, :is_active, expires_on: Date.today) }
+            context "and the 'grace' window is 3 days" do
+              before { allow(Rails.configuration).to receive(:grace_window).and_return(3) }
 
-            it "cannot be renewed" do
-              expect(registration.metaData).to_not allow_event :renew
+              let(:registration) { build(:registration, :is_active, expires_on: Date.today) }
+
+              it "cannot be renewed in 3 days time" do
```

Comment:
Maybe this is personal preference, but I think this might be clearer in a separate context – I'd prefer "when the registration expiration date was three days ago, it cannot be renewed today" to "when the registration expiration date is today, it cannot be renewed in 3 days' time".